src/pykeen/evaluation/utils.py

```diff
@@ -93,3 +95,29 @@ def add(
             description=description,
             link=link,
         )
+
+
+def construct_indicator(*, y_score: np.ndarray, y_true: np.ndarray) -> np.ndarray:
+    """Construct binary indicators from a list of scores.
+
+    If there are $n$ positively labeled entries in ``y_true``, this function
+    assigns the top $n$ highest scores in ``y_score`` as positive and remainder
+    as negative.
+
+    :param y_score:
+        A 1-D array of the score values
+    :param y_true:
+        A 1-D array of binary values (1 and 0)
+    :return:
+        A 1-D array of indicator values
+
+    .. seealso::
+
+        This implementation was inspired by
+        https://github.com/xptree/NetMF/blob/77286b826c4af149055237cef65e2a500e15631a/predict.py#L25-L33
+    """
+    number_pos = np.sum(y_true, dtype=int)
+    y_sort = np.flip(np.argsort(y_score))
```

Comment:
should we add a flag `increasing` so we can have each model say if bigger scores are better?

Reply:
For each model, we already require larger scores to be better (this also has implications, e.g., for losses, etc).

Reply:
Alright then this isn't a concern